### ISSUE TITLE
recharts@npm:2.1.0 doesn't provide prop-types, requested by react-smooth

### ISSUE THREAD

**Issue by wojtekmaj** [recharts/recharts#2627]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
n/a

### Steps to reproduce
* Install recharts using Yarn 2 or 3

### What is expected?
No warnings during installation

### What is actually happening?
Warnings during installation: 

```
% yarn
➤ YN0000: ┌ Resolution step
➤ YN0002: │ recharts@npm:2.1.0 [02bd8] doesn't provide prop-types (p575f4), requested by react-smooth
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed in 0s 510ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 0s 546ms
```

```
% yarn explain peer-requirements p575f4
➤ YN0000: recharts@npm:2.1.0 [02bd8] doesn't provide prop-types, breaking the following requirements:

➤ YN0000: react-smooth@npm:2.0.0 [1cbc0] → ^15.6.0 ✘
```

| Environment | Info |
|---|---|
| Recharts | v2.0.8 |
| React | v17.0.2 |
| System | macOS X |
| Browser | Chrome |

**Comment by wojtekmaj** [comment]
How to fix:
Add prop-types@^15.6.0 to your peerDependencies.

**Comment by RobTheFiveNine** [comment]
> **How to fix this:** Add prop-types@^15.6.0 to your peerDependencies.
> 
> **How to work around this issue if you're using Yarn:** Add the following to `yarnrc.yml`:
> 
> ```
> packageExtensions:
>   recharts@*:
>     peerDependencies:
>       prop-types: ^15.6.0
> ```

Thanks for the workaround
